### PR TITLE
Add HostRating

### PR DESCRIPTION
**Pull request by kllysng**
I added HostRating to the sql create/select files, migration_config, and custom_functions. 

There are errors when I attempt to run migrate.py, I am not sure if it stems from HostRating ID autoincrementing, or not. And would appreciate some insight. Maybe it runs fine on your machine?

Thank you!!

### CODE REVIEW

File: custom_functions.py [hunk 1]
```diff
@@ -31,6 +31,13 @@ def generate_password():
     characters = string.ascii_letters + string.digits
     return ''.join(random.choice(characters) for _ in range(12))
 
+# Custom Action for Token: {{GENERATE_PASSWORD}}
+id = 0
+def generate_id():
+    global id
+    result = id + 1
+    id = result
+    return result
```
Comment (ibsenc) [inline custom_functions.py:40]: Happy to see you making a custom function! But this one won't be necessary as the DB auto increments the primary key for us.

File: custom_functions.py [hunk 2]
```diff
@@ -43,5 +50,7 @@ def craft_value(token, table_column_to_csv_value):
         return generate_username(table_column_to_csv_value)
     if token == "{{GENERATE_PASSWORD}}":
         return generate_password()
+    if token == "{{GENERATE_ID}}":
+        return generate_id()
```
Comment (ibsenc) [inline custom_functions.py:54]: Won't be needed, as per other comment.

File: migration_config.py [hunk 3]
```diff
@@ -64,6 +64,23 @@
...
                     },
                 ]
+            },
+            {
+                "name": "HostRating",
+                "mappings": [
+                    {
+                        "table_column": "ID",
+                        "csv_column": "{{GENERATE_ID}}"
+                    },
```
Comment (ibsenc) [inline migration_config.py]: Automatically added by the DB, please remove

File: migration_config.py [hunk 4]
```diff
@@ -143,6 +160,9 @@
     },
     "Review": {
         "ID": set()
+    },
+    "HostRating": {
+        "ID": set()
```
Comment (ibsenc) [inline migration_config.py:161]: Please remove, as ID won't be processed by this script

File: sql_files/create_tables.sql [hunk 5]
```diff
@@ -71,6 +71,16 @@ CREATE TABLE Review (
         ON UPDATE CASCADE ON DELETE CASCADE */
 );
 
+# Create Tabe HostRaing
+CREATE TABLE HostRating (
+	ID INT UNSIGNED AUTO_INCREMENT,
```
Comment (ibsenc) [inline sql_files/create_tables.sql:76]: This looks good! And also why we can rely on the DB to make it for us.

**Review by ibsenc — CHANGES_REQUESTED** [review]
Looks great, please update base on comments to address the ID issue.

**Comment by kllysng** [comment]
The fix took me a long time because I had to step through and learn the migrate.py code - the issue was that on line 67 in migrate.py, it adds empty (null) values as "", instead of null. An error would occur on the first Rating entry where the csv had an empty value, as it would attempt to add an empty string (instead of null) to the table, instead of a compatible data type (null or decimal in this case). 😊

**Comment by ibsenc** [comment]
Your changes have been tested and integrated with the main branch. Thanks!

**Pull request closed without merging**